### PR TITLE
update dependency version

### PR DESCRIPTION
**Pull request by kazrael2119**
fix https://github.com/Azure/autorest.typescript/issues/2167
update sinon, directory-tree puppeteer versions to fix this issue

### CODE REVIEW

File: packages/autorest.typescript/package.json [hunk 1]
```diff
@@ -81,15 +81,15 @@
     "@azure/abort-controller": "^1.0.1",
     "@azure/core-util": "^1.4.0",
     "@azure/core-xml": "^1.0.0-beta.1",
-    "@microsoft.azure/autorest.testserver": "^3.3.34",
+    "@microsoft.azure/autorest.testserver": "^3.3.48",
```
Comment (MaryGao) [inline packages/autorest.typescript/package.json:84]: why upgrade this?

File: packages/autorest.typescript/test/rlcIntegration/lroRest.spec.ts [hunk 2]
```diff
@@ -1001,7 +1001,7 @@ describe("LRO Rest Client", () => {
       assert.equal(result.status, "200");
     });
 
-    it("should handle put200InvalidJson", async () => {
+    it.only("should handle put200InvalidJson", async () => {
```
Comment (kazrael2119) [inline packages/autorest.typescript/test/rlcIntegration/lroRest.spec.ts:1004]: image
repro ci test failure in my local env.

step:
1. start a **git bash**
2. run ```rush update && rush build```
3. ```cd package/autorest.typescript```
4. run ```rushx start-test-server:v2```
5. start a new **git bash** again
6. ```cd package/autorest.typescript```
7. run ```rushx rlc-integration-test:alone```
8. run ```rushx stop-test-server```

File: packages/autorest.typescript/package.json [hunk 3]
```diff
@@ -109,10 +109,10 @@
     "node-cmd": "^3.0.0",
     "npm-run-all": "^4.1.5",
     "path-browserify": "^1.0.1",
-    "puppeteer": "^3.3.0",
+    "puppeteer": "^21.6.0",
```
Comment (qiaozha) [inline packages/autorest.typescript/package.json:112]: Can you check on this.

Comment (kazrael2119) [reply packages/autorest.typescript/package.json:112]: in main, it's ```"^21.0.0"```

**Comment by qiaozha** [comment]
can we close this as of #2357 ?

**Comment by qiaozha** [comment]
close it for now

**Pull request closed without merging**